stp_core/common/logging/TimeAndSizeRotatingFileHandler.py

```diff
@@ -30,11 +30,48 @@ def rotation_filename(self, default_name: str):
         maxIndex = -1
         for fileName in fileNames:
             if fileName.startswith(defaultFileName):
-                split = fileName.split(".")
-                try:
-                    index = int(split[-1] if len(split) > 0 else 0)
-                except ValueError:
-                    index = 0
+                index = self._file_index(fileName)
                 if index > maxIndex:
                     maxIndex = index
         return "{}.{}".format(default_name, maxIndex + 1)
+
+    @staticmethod
+    def _file_index(file_name):
+        split = file_name.split(".")
+        try:
+            return int(split[-1] if len(split) > 0 else 0)
```

Comment:
Yes, you are right, this code existed before but i copied it at another place without thinking.